components/ui/button/CustomButton.tsx

```diff
@@ -1,56 +1,94 @@
 import React from 'react';
 import { Text, TouchableOpacity, View } from 'react-native';
 
+import { Availability } from '@/util/Availability';
+
 interface CustomButtonProps {
   title: string;
   onPress: () => void;
   disabled?: boolean;
-  type?: 'primary' | 'reversed';
+  type?: ButtonType;
+  size?: ButtonSize;
+}
+
+export enum ButtonType {
+  PRIMARY = 'PRIMARY',
+  OUTLINED = 'OUTLINED',
+}
+
+export enum ButtonSize {
+  SMALL = 'SMALL',
+  LARGE = 'LARGE',
+  BLOCK = 'BLOCK',
 }
 
 const CustomButton: React.FC<CustomButtonProps> = ({
   title,
   onPress,
-  disabled,
-  type = 'primary',
+  disabled = false,
+  type = ButtonType.PRIMARY,
+  size = ButtonSize.BLOCK,
 }) => {
-  const backgroundColor = function (type: 'primary' | 'reversed', disabled?: boolean): string {
-    if (disabled) {
-      if (type === 'primary') return 'bg-sky-light';
-      return 'bg-sky-lightest';
-    }
-    if (type === 'primary') return 'bg-primary-base';
-    return 'bg-sky-lightest';
-  };
+  function backgroundColor(): string {
+    const disabledType = `${disabled ? Availability.DISABLED : Availability.ENABLED}-${type}`;
 
-  const borderColor = function (type: 'primary' | 'reversed', disabled?: boolean): string {
-    if (disabled) {
-      return 'border-sky-light';
+    switch (disabledType) {
+      case `${Availability.DISABLED}-${ButtonType.PRIMARY}`:
+        return 'bg-sky-light dark:bg-ink-dark';
+      case `${Availability.DISABLED}-${ButtonType.OUTLINED}`:
+        return 'bg-sky-latest dark:bg-ink-darkest';
+      case `${Availability.ENABLED}-${ButtonType.PRIMARY}`:
+        return 'bg-primary-base';
+      case `${Availability.ENABLED}-${ButtonType.OUTLINED}`:
+        return 'bg-sky-latest dark:bg-ink-darkest';
+      default:
+        return '';
     }
-    return 'border-primary-base';
-  };
+  }
+
+  function borderColor(): string {
+    const disabledType = `${disabled ? Availability.DISABLED : Availability.ENABLED}-${type}`;
 
-  const textColor = function (type: 'primary' | 'reversed', disabled?: boolean): string {
-    if (disabled) {
-      if (type === 'primary') return 'text-sky-dark';
-      return 'text-sky-light';
+    switch (disabledType) {
+      case `${Availability.DISABLED}-${ButtonType.OUTLINED}`:
+        return 'border-sky-base dark:border-ink-base';
+      case `${Availability.ENABLED}-${ButtonType.OUTLINED}`:
+        return 'border-primary-base dark:border-primary-light';
+      default:
+        return '';
     }
-    if (type === 'primary') return 'text-sky-lightest';
-    return 'text-primary-base';
-  };
+  }
 
+  function textColor(): string {
+    const disabledType = `${disabled ? Availability.DISABLED : Availability.ENABLED}-${type}`;
```

Comment:
i see 3 times that function to check if it disabled maybe extract that to function ?